onnx/test/training_tool_test.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+import onnx
+import numpy as np
+from onnx.tools import update_model_dims
+from onnx import helper, numpy_helper, TensorProto
+
+
+class TestTrainingTool(unittest.TestCase):
+    def test_training_info_proto(self):  # type: () -> None
+        # Inference graph.
+        A_shape = [2, 2]
+        A_name = 'A'
+        A = np.array(A_shape).astype(np.float)
+        A_initializer = numpy_helper.from_array(A, name=A_name)
+        A_value_info = helper.make_tensor_value_info(
+            A_name, TensorProto.FLOAT, A_shape)
+
+        B_shape = [2, 2]
+        B_name = 'B'
+        B = np.array(B_shape).astype(np.float)
+        B_initializer = numpy_helper.from_array(B, name=B_name)
+        B_value_info = helper.make_tensor_value_info(
+            B_name, TensorProto.FLOAT, B_shape)
+
+        C_shape = [2, 2]
+        C_name = 'C'
+        C_value_info = helper.make_tensor_value_info(
+            C_name, TensorProto.FLOAT, C_shape)
+
+        inference_node = helper.make_node(
+            'MatMul',
+            inputs=[A_name, B_name],
+            outputs=[C_name]
+        )
+
+        inference_graph = helper.make_graph(
+            [inference_node],
+            'simple_inference',
+            [A_value_info, B_value_info],
+            [C_value_info],
+            [A_initializer, B_initializer]
+        )
+
+        # Training graph
+        X_shape = [2, 2]
+        X_name = 'X'
+        X = np.array(X_shape).astype(np.float)
+        X_initializer = numpy_helper.from_array(X, name=X_name)
+        X_value_info = helper.make_tensor_value_info(
+            X_name, TensorProto.FLOAT, X_shape)
+
+        Y_shape = [2, 2]
+        Y_name = 'Y'
+        Y_value_info = helper.make_tensor_value_info(
+            Y_name, TensorProto.FLOAT, Y_shape)
+
+        node = helper.make_node(
+            'MatMul',
+            inputs=[X_name, C_name],  # tensor "C" is from inference graph.
+            outputs=[Y_name]
+        )
+
+        training_graph = helper.make_graph(
+            [node],
+            'simple_training',
+            [X_value_info],
+            [Y_value_info],
+            [X_initializer]
+        )
+
+        # Capture assignment of B <--- Y.
+        training_info = helper.make_training_info(
+            training_graph, [(B_name, Y_name)], None, None)
+
+        # Create a model with both inference and training information.
+        model = helper.make_model(inference_graph)
+        # Check if the inference-only part is correct.
+        onnx.checker.check_model(model)
+        # Insert training information.
+        new_training_info = model.training_info.add()
+        new_training_info.CopyFrom(training_info)
```

Comment:
After this line, `print (model)` prints this model in text format. It should show inference graph + training graph.